packages/react-components/src/components/table-output-component.tsx

```diff
@@ -626,13 +639,28 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
     private async findMatchedEvent(direction: Direction) {
         let isFound = false;
         if (this.gridApi) {
+            const msBetweenChecks = 100;
+            while (this.gridMatched) {
+                // wait for grid to be done being matched -from elsewhere (concurrently)
+                await new Promise(cb => setTimeout(cb, msBetweenChecks));
+            }
+            while (this.gridRedrawn) {
+                // wait for grid to be done being redrawn -elsewhere (before assuming it)
+                await new Promise(cb => setTimeout(cb, msBetweenChecks));
+            }
```

Comment:
I couldn't trigger these awaits. I'm afraid they are never called.

Reply:
I propose to keep this in as (small) safety, until we can work on the more major refactoring you proposed.